src/benchmarklib/benchmark_utils.cpp

```diff
@@ -31,28 +31,41 @@ std::ostream& get_out_stream(const bool verbose) {
 }
 
 BenchmarkState::BenchmarkState(const size_t max_num_iterations, const opossum::Duration max_duration)
-    : max_num_iterations(max_num_iterations), max_duration(max_duration) {}
+    : max_num_iterations(max_num_iterations), max_duration(max_duration) {
+  iteration_durations.reserve(max_num_iterations);
+}
 
 bool BenchmarkState::keep_running() {
+  bool is_first_iteration = false;
+
   switch (state) {
     case State::NotStarted:
-      begin = std::chrono::high_resolution_clock::now();
+      benchmark_begin = std::chrono::high_resolution_clock::now();
       state = State::Running;
+      is_first_iteration = true;
       break;
     case State::Over:
       return false;
     default: {}
   }
 
+  const auto now = std::chrono::high_resolution_clock::now();
+  const auto benchmark_duration = now - benchmark_begin;
+  const auto iteration_duration = now - iteration_begin;
+
+  if (!is_first_iteration) {
+    iteration_durations.push_back(iteration_duration);
+  }
+
+  benchmark_end = now;
```

Comment:
> What is a good lower boundary for required runs before claiming significance? I guess 100 would be nice, but with scale factor >=1.0, ain't nobody got time for that. 10?

10 sounds good. We should also make sure that the benchmark ran for at least... I don't know... one minute?

Reply:
```

+-----------+-------------------+------+-------------------+------+--------+---------------------------------+
| Benchmark | prev. iter/s      | runs | new iter/s        | runs | change | p-value (significant if <0.001) |
+-----------+-------------------+------+-------------------+------+--------+---------------------------------+
| TPC-H 1   | 0.842009425163269 | 25   | 0.843953311443329 | 25   | +0%    |     (run time too short) 0.2808 |
| TPC-H 3   | 8.33191013336182  | 25   | 8.03579711914062  | 25   | -4%    |     (run time too short) 0.0000 |
| TPC-H 5   | 2.15718054771423  | 25   | 2.4319965839386   | 25   | +13%   |     (run time too short) 0.0041 |
| TPC-H 6   | 19.5317764282227  | 25   | 19.5994606018066  | 25   | +0%    |     (run time too short) 0.5775 |
| TPC-H 7   | 1.62585973739624  | 25   | 1.53261685371399  | 25   | -6%    |     (run time too short) 0.0000 |
| TPC-H 9   | 0.15766803920269  | 15   | 0.160950303077698 | 10   | +2%    |                          0.0009 |
| TPC-H 10  | 5.89235353469849  | 25   | 7.12257146835327  | 25   | +21%   |     (run time too short) 0.0000 |
| TPC-H 13  | 10.8946914672852  | 25   | 10.6563177108765  | 25   | -2%    |     (run time too short) 0.0817 |
| average   |                   |      |                   |      | +3%    |                                 |
+-----------+-------------------+------+-------------------+------+--------+---------------------------------+
```
(sorry no screenshot, I messed up my terminal's font color palette and right now grey is indistinguishable from black :sweat_smile: )